conda/resolve.py

```diff
@@ -500,18 +502,24 @@ def gen_clauses(self):
             for ms in self.ms_depends(prec):
                 C.Require(C.Or, nkey, self.push_MatchSpec(C, ms))
 
-        log.debug("gen_clauses returning with clause count: %s", len(C.clauses))
+        if log.isEnabledFor(DEBUG):
```

Comment:
As written above, this guard is just because `C.get_clause_count()` might be costly if we don't save the clause count explicitly anywhere, like is the case with `common.logic.ClauseArray`.